### PR TITLE
fix ec2 join check

### LINKED ISSUE #13677: `unsupported role: "Instance"` errors with EC2 join method
Expected behavior:
Nodes can use the EC2 join method with no error logs.

Current behavior:
The following is logged once per minute:
```
2022-06-21T01:34:42Z INFO [PROC:1]    Joining the cluster with a secure token. pid:10387.1 service/connect.go:576
2022-06-21T01:34:42Z INFO [AUTH]      Attempting registration via proxy server. auth/register.go:177
2022-06-21T01:34:42Z INFO [AUTH]      Attempting registration with auth server. auth/register.go:177
2022-06-21T01:34:42Z WARN [AUTH]      Joining cluster without validating the identity of the Auth Server. This may open you up to a Man-In-The-Middle (MITM) attack if an attacker can gain privileged network access. To remedy this, use the CA pin value provided when join token was generated to validate the identity of the Auth Server. auth/register.go:333
2022-06-21T01:34:42Z ERRO [PROC:1]    Instance failed to establish connection to cluster: unsupported role: "Instance", Post "https://teleport.cluster.local/v2/tokens/register": remote error: tls: internal error. pid:10387.1 service/connect.go:106
```

Bug details:
- Teleport version
version 10
- Recreation steps
join a node/app/db/kube to the cluster using the EC2 join method https://goteleport.com/docs/setup/guides/joining-nodes-aws-ec2/
- Debug logs
above

@fspmarshall I'm not sure what the "Instance" role does or if this currently breaks anything, but we'll need to fix this, the logs are not nice and it's failing to get certs for the "Instance" role. The EC2 join method does a uniqueness check [here](https://github.com/gravitational/teleport/blob/ca520999c1f3e929e98f37c551532446bfbfbbd7/lib/auth/join_ec2.go#L261) and does not know how to check if an "Instance" is unique.

### PR DESCRIPTION
**Pull request by fspmarshall**
Fixes https://github.com/gravitational/teleport/issues/13677

### CODE REVIEW

File: lib/auth/join_ec2.go [hunk 1]
```diff
@@ -279,6 +277,9 @@ func (a *Server) checkInstanceUnique(ctx context.Context, req *types.RegisterUsi
 		instanceExists, err = appExists(ctx, a, req.HostID)
 	case types.RoleDatabase:
 		instanceExists, err = dbExists(ctx, a, req.HostID)
+	case types.RoleInstance:
+		// no appropriate check exists for the Instance role
+		instanceExists = true
```
Comment (nklaassen) [inline lib/auth/join_ec2.go:282]:
```suggestion
		instanceExists = false
```

**Review by nklaassen — CHANGES_REQUESTED** [review]

File: lib/auth/join_ec2.go (hunk 1 above)
Comment (fspmarshall) [reply lib/auth/join_ec2.go:282]: lmao, one sec.

**Review by nklaassen — APPROVED** [review]

**Comment by zmb3** [comment]
Can we add test coverage to prevent future regressions?

**Comment by fspmarshall** [comment]
@zmb3 Catching this kind of regression requires proper integration testing.  We have integration tests that would have caught this, but it seems they aren't run because they rely on the real EC2 api.

**Comment by nklaassen** [comment]
> Catching this kind of regression requires proper integration testing. We have integration tests that would have caught this, but it seems they aren't run because they rely on the real EC2 api.

fwiw that test still passes and nothing seems broken, I only found this because I noticed the logs

**Review by rosstimothy — APPROVED** [review]

**Comment by fspmarshall** [comment]
> fwiw that test still passes and nothing seems broken, I only found this because I noticed the logs

@nklaassen ah, didn't realize that.  Looks like test was able to pass if the ssh service successfully heartbeats, even if the node couldn't finish all startup steps. Updated the existing integration test to now require that the node actually emits `TeleportReady` and confirmed that it fails without the fix in this PR.  This will ensure that going forward, test doesn't pass unless all required certs are successfully acquired.

File: integration/ec2_test.go [hunk 2]
```diff
@@ -198,6 +198,14 @@ func TestEC2NodeJoin(t *testing.T) {
 	require.NoError(t, nodeSvc.Start())
 	t.Cleanup(func() { require.NoError(t, nodeSvc.Close()) })
 
+	eventCh := make(chan service.Event, 1)
+	nodeSvc.WaitForEvent(nodeSvc.ExitContext(), service.TeleportReadyEvent, eventCh)
+	select {
+	case <-eventCh:
+	case <-time.After(10 * time.Second):
+		t.Fatal("timeout waiting for node readyness")
```
Comment (zmb3) [inline integration/ec2_test.go:206]:
```suggestion
		t.Fatal("timeout waiting for node readiness")
```

**Pull request merged**